feat: display all child tags in the "bare bones" taxonomy detail page [FAL-3529] [FC-0036]

bradenmacdonald opened the pull request:
This is a minor improvement to https://github.com/openedx/frontend-app-course-authoring/pull/655

It sort of depends on https://github.com/openedx/openedx-learning/pull/119 but it can be merged even before that one is merged.

This is **not** the final UI design and all of this can/will be changed - this just makes sure that users can see _all_ the tags in the taxonomy during our initial round of testing.

## Before:

![Screenshot 2023-11-17 at 3 22 16 PM](https://github.com/open-craft/frontend-app-course-authoring/assets/945577/e1bf1b08-d2ae-4074-8bd7-0b3960650e77)

## After:

![Screenshot 2023-11-17 at 3 23 01 PM](https://github.com/open-craft/frontend-app-course-authoring/assets/945577/045b50dc-6a29-41b0-acdb-e543cdc7272c)

![Screenshot 2023-11-17 at 3 23 16 PM](https://github.com/open-craft/frontend-app-course-authoring/assets/945577/71ad6591-e02e-4c43-83af-e57062d498c1)

![Screenshot 2023-11-17 at 3 23 30 PM](https://github.com/open-craft/frontend-app-course-authoring/assets/945577/316e2977-0761-4ed9-9724-f382dd3fe59f)

bradenmacdonald commented:
@pomegranited @rpenido FYI - I don't want to delay https://github.com/openedx/frontend-app-course-authoring/pull/655 so I've made this as a separate PR, but this is an improvement I want to make before we update our sandbox and get users testing it.

rpenido commented:
Hi @bradenmacdonald!

LGTM. Just need to fix some lint issues to make the tests pass.

I found a issue using this code to show the taxonomy imported using our template file: it shows the children (all depths) alongside the parents.
![image](https://github.com/open-craft/frontend-app-course-authoring/assets/849463/39b0fce6-26e3-4df0-9ebf-f85d4d605233)
```json
{
  "results":[
    {
      "value":"Wind instruments",
      "external_id":"WINDS",
      "child_count":2,
      "depth":0,
      "parent_value":null,
      "_id":23164,
      "sub_tags_url":"http://localhost:18010/api/content_tagging/v1/taxonomies/19/tags/?parent_tag=Wind%20instruments"
    },
    {
      "value":"Brass",
      "external_id":"BRASS",
      "child_count":2,
      "depth":1,
      "parent_value":"Wind instruments",
      "_id":23174,
      "sub_tags_url":"http://localhost:18010/api/content_tagging/v1/taxonomies/19/tags/?parent_tag=Brass"
    },
    {
      "value":"Trumpet",
      "external_id":"TRUMPET",
      "child_count":0,
      "depth":2,
      "parent_value":"Brass",
      "_id":23186,
      "sub_tags_url":null
    },
    {
      "value":"Tuba",
      "external_id":"TUBA",
      "child_count":0,
      "depth":2,
      "parent_value":"Brass",
      "_id":23187,
      "sub_tags_url":null
    },
    {
      "value":"Woodwinds",
      "external_id":"WOODS",
      "child_count":3,
      "depth":1,
      "parent_value":"Wind instruments",
      "_id":23175,
      "sub_tags_url":"http://localhost:18010/api/content_tagging/v1/taxonomies/19/tags/?parent_tag=Woodwinds"
    },
    {
      "value":"Clarinet",
      "external_id":"CLARINET",
      "child_count":0,
      "depth":2,
      "parent_value":"Woodwinds",
      "_id":23184,
      "sub_tags_url":null
    },
    {
      "value":"Flute",
      "external_id":"FLUTE",
      "child_count":0,
      "depth":2,
      "parent_value":"Woodwinds",
      "_id":23176,
      "sub_tags_url":null
    },
    {
      "value":"Oboe",
      "external_id":"OBOE",
      "child_count":0,
      "depth":2,
      "parent_value":"Woodwinds",
      "_id":23185,
      "sub_tags_url":null
    }
  ]
}
```

I didn't dig too much into it, so the problem is in the backend but I don't know if it is in the `get_filtered_tags` function, the import code or the template itself.

Should we create a task to take a look at this?

rpenido commented on src/generic/Loading.jsx:
```diff
@@ -2,27 +2,30 @@ import React from 'react';
 import { Spinner } from '@edx/paragon';
 import { FormattedMessage } from '@edx/frontend-platform/i18n';
 
+
```
Fix lint
```suggestion
```

rpenido commented on src/taxonomy/tag-list/TagListTable.jsx:
```diff
@@ -1,14 +1,38 @@
+import { useSubTags } from './data/api';
+
+const SubTagsExpanded = ({ taxonomyId, parentTagValue }) => {
+
```
lint 
```suggestion
```

rpenido commented on src/taxonomy/tag-list/TagListTable.jsx:
```diff
@@ -1,14 +1,38 @@
+
+  if (subTagsData.isLoading) {
+    return <LoadingSpinner />;
+  } else if (subTagsData.isError) {
```
lint 
```suggestion
  }
  
  if (subTagsData.isError) {
```

rpenido reviewed (approved):
LGTM :+1: 
* [x]  I read through the code
* [x]  I followed the test instructions and ran the code locally
* [x]  I checked for accessibility issues by navigating using the keyboard only
* [x] User-facing strings are extracted for translation

@bradenmacdonald 
Do you think it is better to merge in the branch opencraft branch now or wait https://github.com/openedx/frontend-app-course-authoring/pull/655 get merged and then create a new PR against `master`?

bradenmacdonald commented:
@rpenido Thanks! I'll wait until your PR is merged (as well as https://github.com/openedx/openedx-learning/pull/119 ), and then I'll re-submit this PR. I don't want to slow down your PR from getting merged.

rpenido committed:
feat: bare bones taxonomy detail page [FC-0036] (#655)

* feat: System-defined tooltip added

* feat: Taxonomy card menu added. Export menu item added

* feat: Modal for export taxonomy

* feat: Connect with export API

* test: Tests for API and selectors

* feat: Use windows.location.href to call the export endpoint

* test: ExportModal.test added

* style: Delete unnecesary code

* docs: README updated with taxonomy feature

* style: TaxonomyCard updated to a better code style

* style: injectIntl replaced by useIntl on taxonomy pages and components

* refactor: Move and rename taxonomy UI components to match 0002 ADR

* refactor: Move api to data to match with 0002 ADR

* test: Refactor ExportModal tests

* chore: Fix validations

* chore: Lint

* refactor: Moving hooks to apiHooks

* feat: add taxonomy detail page

* fix: address nits in PR review

* refactor: move data/selectors to data/apiHooks

and fix tests to mock useQuery.

* fix: address nits in PR review

* fix: replace taxonomy menu ModalPopup with Dropdown menu

Avoids clicking through to the card when using the menu button to hide
a card's menu.

* fix: change taxonomy URLs

* /taxonomy-list is now /taxonomies, and there's a temporary redirect
* /taxonomy-list/:id: is now /taxonomy/:id:

---------

Co-authored-by: Christofer <christofer@opencraft.com>
Co-authored-by: XnpioChV <xnpiochv@gmail.com>
Co-authored-by: Christofer Chavez <christofer@example.com>
Co-authored-by: Jillian Vogel <jill@opencraft.com>
Co-authored-by: Braden MacDonald <braden@opencraft.com>

bradenmacdonald committed:
feat: display all child tags in the "bare bones" taxonomy detail page

bradenmacdonald committed:
feat: set <title> on taxonomy list page and taxonomy detail page

bradenmacdonald commented:
@rpenido Now that your original PR has merged, I have a new PR. Can you please review it there? https://github.com/openedx/frontend-app-course-authoring/pull/703
